osf/migrations/0016_backfill_date_retracted.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-03-14 14:41
+
+"""
+Backfill Retraction.date_retracted with `RETRACTION_APPROVED` log date.
+"""
+
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from osf.models import Registration, Retraction, Sanction
+
+
+def set_date_retracted(*args):
+    registrations = (
+        Registration.objects.filter(retraction__state=Sanction.APPROVED, retraction__date_retracted=None)
+        .select_related('retraction')
```

Comment:
Ah, good call. You are correct. I didn't even look. No-need. `_id` is a local field for `ObjectIDMixin`.